lib/logger/lib/logger/backends/console.ex

```diff
@@ -12,23 +12,42 @@ defmodule Logger.Backends.Console do
   end
 
   def init({device, opts}) do
-    {:ok, configure(device, opts)}
+    {:ok, {configure(device, opts), %{}}}
   end
 
-  def handle_call({:configure, options}, state) do
-    {:ok, :ok, configure(state.device, options)}
+  def handle_call({:configure, options}, {state, refs}) do
+    {:ok, :ok, {configure(state.device, options), refs}}
   end
 
-  def handle_event({_level, gl, _event}, state)
-  when node(gl) != node() do
-    {:ok, state}
+  def handle_event({_level, gl, _event}, {_, _} = pair) when node(gl) != node() do
+    {:ok, pair}
   end
 
-  def handle_event({level, _gl, {Logger, msg, ts, md}}, state) do
+  def handle_event({level, _gl, {Logger, msg, ts, md}}, {state, refs} = pair) do
     if meet_level?(level, state.level) do
-      log_event(level, msg, ts, md, state)
+      ref = log_event(level, msg, ts, md, state)
+      {:ok, {state, Map.put(refs, ref, {ts, msg})}}
+    else
+      {:ok, pair}
+    end
+  end
+
+  def handle_info({:io_reply, ref, status}, {state, refs} = pair) do
+    case refs do
+      %{^ref => _} when status == :ok ->
+        {:ok, {state, Map.delete(refs, ref)}}
+      %{^ref => {ts, msg}} ->
+        # We use width infinity because the message has already been truncated
+        log_event(:error, ["Failure while logging console message: ",
```

Comment:
@fishcakez I thought about spawning a new process here... but I thought it is probably fine to keep things as is.